lib/react.dart

```diff
@@ -5,14 +5,56 @@
 /// A Dart library for building UI using ReactJS.
 library react;
 
-/// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/top-level-api.html#react.component)
-/// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/component-api.html)
+import 'package:react/src/typedefs.dart';
+
+/// Top-level ReactJS [Component class](https://facebook.github.io/react/docs/react-component.html)
+/// which provides the [ReactJS Component API](https://facebook.github.io/react/docs/react-component.html#reference)
 abstract class Component {
-  /// ReactJS `Component` props.
-  Map props;
+  /// ReactJS [Component] props.
+  ///
+  /// Currently private since the `@virtual` annotation within the meta package
+  /// [doesn't work for overriding fields](https://github.com/dart-lang/sdk/issues/27452).
+  ///
+  /// TODO: Make public once SDK issue / meta package allows field overriding in strong mode.
+  ///
+  /// See: [props]
+  ///
+  /// Related: [_state]
+  Map _props;
+
+  /// ReactJS [Component] state.
+  ///
+  /// Currently private since the `@virtual` annotation within the meta package
+  /// [doesn't work for overriding fields](https://github.com/dart-lang/sdk/issues/27452).
+  ///
+  /// TODO: Make public once SDK issue / meta package allows field overriding in strong mode.
+  ///
+  /// See: [state]
+  ///
+  /// Related: [_props]
+  Map _state = {};
 
-  /// Provides access to the underlying DOM representation of the [render]ed `Component`.
-  dynamic ref;
+  /// Returns a component reference.
+  ///
+  /// * [Component] if it is a Dart component.
+  /// * `Element` _(DOM node)_ if it is a React DOM component.
+  ///
+  /// Currently private since the `@virtual` annotation within the meta package
+  /// [doesn't work for overriding fields](https://github.com/dart-lang/sdk/issues/27452).
+  ///
+  /// TODO: Make public once SDK issue / meta package allows field overriding in strong mode.
+  ///
+  /// See: [ref]
+  Ref _ref;
+
+  Map get props => _props;
+  set props(Map value) => _props = value;
+
+  Map get state => _state;
+  set state(Map value) => _state = value;
+
+  Ref get ref => _ref;
```

Comment:
These getters should have doc comments, since consumers can't see the doc comments of private fields.

What about this:
```dart
/// A private field that backs [ref], which is exposed via getter/setter so
/// it can be overridden in strong mode.
///
/// Necessary since the `@virtual` annotation within the meta package
/// [doesn't work for overriding fields](https://github.com/dart-lang/sdk/issues/27452).
///
/// TODO: Switch back to a plain field once this issue is fixed.
Ref _ref;

...

/// A function that returns a component reference:
///
/// * [Component] if it is a Dart component.
/// * `Element` _(DOM node)_ if it is a React DOM component.
Ref get ref => _ref;
set ref(Ref value) => _ref = value;
```